tests/integration/conftest.py

```diff
@@ -50,47 +51,65 @@ async def app(ops_test: OpsTest, app_name: str):
 
     # Add required relations
     await ops_test.model.add_relation(hello_kubecon_app_name, app_name)
-    await ops_test.model.wait_for_idle(status=ActiveStatus.name, idle_period=120)
+    await ops_test.model.wait_for_idle(timeout=10 * 60)
 
     yield application
 
 
 @pytest_asyncio.fixture(scope="module")
-async def app_ip(app: Application):
-    """Get unit IP address and add to /etc/hosts."""
-    # Get the IP address which is in the status message
+async def ip_address_list(ops_test: OpsTest, app: Application):
+    """Get unit IP address from workload message.
+
+    Example: Ingress IP(s): 127.0.0.1, Service IP(s): 10.152.183.84
+    """
+    # Reduce the update_status frequency until the cluster is deployed
+    async with ops_test.fast_forward():
+        await ops_test.model.block_until(
+            lambda: "Ingress IP(s)" in app.units[0].workload_status_message, timeout=100
```

Comment:
That is a nice implementation, I wasn't aware of that

Reply:
I found while was researching about the "Task was destroyed..." error :)